src/src/com/microsoft/aad/adal/DateTimeAdapter.java

```diff
@@ -50,12 +50,31 @@
             DateFormat.DEFAULT);
 
     private final DateFormat iso8601Format = buildIso8601Format();
```

Comment:
Users could stay on old version of onedrive, a version that's one year old. I don't expect so, but it could happen, and they update it, i.e. recently, Then I can understand that the exception is thrown. Otherwise, i don't know why we get the exception in onedrive. If you get more data around it, feel free to let me know. Or I can play with your telemetry data.